src/hooks.js

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
+    if (state.response.status !== 204) {
+      state.body = await state.response.text()
+    }
+    state.error = state.response.ok
+      ? undefined
+      : Error(state.response.statusText)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetchJSON = (...args) => {
+  const state = useFetch(...args)
+  if (!state.body) return state
+  try {
+    state.data = JSON.parse(state.body)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetch = (url, fetchOpts, inputs = [url]) => {
+  const [state, setState] = useState({ pending: true })
+  // biome-ignore lint/correctness/useExhaustiveDependencies: dependencies defined by the caller
+  useEffect(() => {
+    if (url == null) return
```

Comment:
We may need to call the hook without fetching data yet, like, we are getting data from the gsheet first and all or paperdoll items don't have an ID yet, we still want to call the hook.

This is one of the downside of the react hook api, you need to call always the same number of hooks because react need the order of the hooks to retrigger component rendering, it's a bit magic and painfull at the sametime.

So we need a way to call the hook waiting for the url